### PR TITLE
currency changing dropdown

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

**Comment by ScottDowne** [comment]
R?

File: pages/sequential.jsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import Footer from '../components/footer.jsx';
 import Header from '../components/header.jsx';
 import SmallPrint from '../components/small-print.jsx';
 import SectionHeading from '../components/section-heading.jsx';
+import CurrencyDrop from '../components/currency-dropdown.jsx';
```
Comment (alicoding) [inline pages/sequential.jsx]: Can you rename this to `CurrencyDropDown` instead?

File: pages/single-form.jsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import Footer from '../components/footer.jsx';
 import Header from '../components/header.jsx';
 import SmallPrint from '../components/small-print.jsx';
 import SectionHeading from '../components/section-heading.jsx';
+import CurrencyDrop from '../components/currency-dropdown.jsx';
```
Comment (alicoding) [inline pages/single-form.jsx]: And here

File: components/currency-dropdown.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+module.exports = React.createClass({
+  render: function() {
+    var currencies = this.props.currencies;
+    return (
+      <select onChange={this.props.onChange} className="currency-dropdown" value={this.props.currency}>
+        {Object.keys(currencies).map(function(currency, i) {
+          return (
+            <option value={currency} key={i}>
+              {currency + " " + currencies[currency].symbol}
```
Comment (alicoding) [inline components/currency-dropdown.jsx]: I don't think we need a symbol beside the currency code here.

Comment (alicoding) [inline components/currency-dropdown.jsx]: We either going to just show currency code alone or showing something like this:

`Canadian Dollar (CAD)`

**Comment by alicoding** [comment]
There is a bug when you pick a `NaN` and go back to reselect a valid number it doesn't seem to update the amount when trying to submit again. Here is the STR:

1. Set presets in the query string to `presets=1,2,3,a`
2. Select `NaN`
3. Fill in the form and try to donate
4. You should see an error which is expected.
5. Go back to step one of the donation flow and choose a valid amount
6. Try to donate again and it should allow you this time, but it is not.

File: components/currency-dropdown.jsx (hunk 3 above)
Comment (ScottDowne) [inline components/currency-dropdown.jsx]: We also display symbol beside the other amount box. Is that ok there? Is it only the dropdown that is the issue?

**Comment by ScottDowne** [comment]
Updated. r?

**Comment by alicoding** [comment]
Looking good now. Just left you a comment there

File: mixins/form.jsx [hunk 4]
```diff
@@ -272,7 +309,15 @@ module.exports = {
...
     var props = {};
     fields.forEach(function(name) {
-      props = assign(props, self.state[name].state.values);
+      var prop = self.state[name].state.values;
+      // Currently some fields expose their values on the form, and not themselves.
+      // So we need to check for props in both places until all field componenets are updated.
+      if (!prop) {
+        prop = self.state.props[name].values;
+      }
+      if (prop) {
```
Comment (alicoding) [inline mixins/form.jsx]: do you really need this check now? it looks like you are going to have prop no matter what?

**Comment by ScottDowne** [comment]
@alicoding Updated

**Comment by alicoding** [comment]
r+

**Pull request merged**